src/main/java/trackbot/ui/Parser.java

```diff
@@ -0,0 +1,205 @@
+package trackbot.ui;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Pattern;
+
+import trackbot.TrackBotException;
+import trackbot.commands.AddCommand;
+import trackbot.commands.Command;
+import trackbot.commands.DeleteCommand;
+import trackbot.commands.ExitCommand;
+import trackbot.commands.FindCommand;
+import trackbot.commands.ListCommand;
+import trackbot.commands.MarkCommand;
+import trackbot.commands.UnmarkCommand;
+import trackbot.task.Deadline;
+import trackbot.task.Event;
+import trackbot.task.Task;
+import trackbot.task.ToDo;
+
+/**
+ * Parses user inputs and translate them into commands to be executed by trackbot.
+ * The Parser class provides utility methods for parsing tasks from storage for display
+ * and validating date formats.
+ */
+public class Parser {
+
+    static final String DEFAULT_DATE_FORMAT = "MMM d yyyy";
+    static final String DEFAULT_DATE_TIME_FORMAT = "MMM d yyyy HH:mm";
+
+    /**
+     * Parses the user input and returns the corresponding Command.
+     *
+     * @param userInput The raw input string from the user.
+     * @return The Command object representing the user input.
+     * @throws TrackBotException If the input does not match any known commands or has an invalid format.
+     */
+    public Command parse(String userInput) throws TrackBotException {
+        String[] splitInput = userInput.split(" ", 2);
+        String commandWord = splitInput[0].toLowerCase();
+        String otherWord = splitInput.length > 1 ? splitInput[1].trim() : "";
+
+        switch (commandWord) {
```

Comment:
Switch statements shouldn't have indentations. Personally I would have done something like this:

``` 
switch (commandWord) {
case "list": 
if (!otherWord.isEmpty()) {
    throw TrackBotException.invalidFormat("list", "list");
}
return new ListCommand();
.
.
.
```

Reply:
I believe what I did was correct . The below code is taken from the java coding standard given [here](https://se-education.org/guides/conventions/java/intermediate.html)


```ruby
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```

Reply:
You're right, misread the coding standard. Thanks.